### ISSUE TITLE
createdump fails for unknown reason

### ISSUE THREAD

**Issue by andtii** [dotnet/coreclr#25152]
Running a creatdump using the mcr.microsoft.com/dotnet/core/aspnet:2.2.5 image result in a failure.

![image](https://user-images.githubusercontent.com/1434571/59436712-6c4a8000-8df0-11e9-902c-454b5364a85f.png)

**Comment by AaronRobinsonMSFT** [comment]
cc @leculver

**Comment by leculver** [comment]
@mikem8361 @tommcdon Can you guys take a look here?  Thank you!

**Comment by joe-elliott** [comment]
andtii is running createdump according to the guides in here https://github.com/joe-elliott/netcore-kubernetes-profiling

and reported an issue here:

https://github.com/joe-elliott/netcore-kubernetes-profiling/issues/4

I am able to reproduce that error by running the debian createdump against an application running in an alpine container.  Currently waiting on him to post yaml/other details that might help understand why he got this error.

**Comment by andtii** [comment]
This is exactly what we are doing, we have a lot of producion containers running and we want to have a sidecar container that has everything needed to generate the dumps. The sidecar container is debian 2.2.5 and the prod container is the alpine 2.2.5. We have this production issue where something eats up all our memory and i have spent 1 week just trying to get a dump from the production containers so we can investigate what goes wrong and still havent found a good solution. @mikem8361 any ideas how we can progress?

**Comment by mikem8361** [comment]
If I understand correctly, you are in a debian container using the createdump from the debian .NET Core runtime trying to generate a dump in an Alpine container. 

The problem is that debian/alpine versions of the createdump and the DAC (libmscordaccore.so) are not compatible. Debian is built with glibc and Alpine is built with musl libraries. The debian createdump is looking for debian libmscordaccore.so (next to its createdump) which will NOT work with a alpine .NET Core process/container.

Either they both have to be debian based (which also includes ubuntu, fedora, opensuse, etc.) or both MUSL based which Alpine is the only one I know.

**Comment by joe-elliott** [comment]
So I suppose the broader issue here is analysis of coredumps in the Alpine container.  

When I use this base image:

```
FROM mcr.microsoft.com/dotnet/core/aspnet:2.2.5-alpine3.9
```

I can successfully generate a coredump using createdump, and I can install and run lldb, but I am unable to use some of the sos commands:

```
(lldb) sos DumpHeap
Error requesting heap segment 00007F1310BF8000
Failed to retrieve segments for gc heap
Unable to build snapshot of the garbage collector state
DumpHeap  failed
```

lldb version is 5.0.2.  Any advice on how to overcome this?

**Comment by mikem8361** [comment]
We haven't found a version lldb that worked on alpine. Which why we did dotnet-dump analyze where you can run SOS commands on core dumps without lldb. See the diagnostics repo on installation https://github.com/dotnet/diagnostics.git

**Comment by mikem8361** [comment]
More specifically see this doc: https://github.com/dotnet/diagnostics/blob/master/documentation/dotnet-dump-instructions.md#dump-collection-and-analysis-utility-dotnet-dump

**Comment by joe-elliott** [comment]
Nice.  Thank you for the link.

I'm having issues getting `dotnet dump` to work, but that is probably a separate issue.  Filing a ticket with `dotnet/diagnostics`.

**Comment by mikem8361** [comment]
Sorry, I forgot about the Alpine createdump problem that is fixed in 3.0.x (Preview 6) but in 2.2.x. See dotnet/coreclr#24599.

I will see about getting this fix into the next 2.2 service release.

**Comment by noahfalk** [comment]
I think the primary issue was resolved and 2.2 will be out of support in 8 weeks. Closing this.
